src/test/java/eu/elixir/ega/ebi/it/res/FileControllerIntgTest.java

```diff
@@ -0,0 +1,96 @@
+package eu.elixir.ega.ebi.it.res;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+
+import org.apache.http.HttpStatus;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import com.google.common.io.CountingInputStream;
+import com.jayway.restassured.response.Response;
+
+/**
+ * Class to do the integration test for FileController in ega-data-api-key.
+ * 
+ * @author amohan
+ *
+ */
+public class FileControllerIntgTest extends ResBase {
+
+    private final static String FILE_PATH = "/file/archive/";
+    private final static String KEY_ID_ZERO = "000";
+
+    /**
+     * Verify the api call /file/archive/{id} and check status is
+     * {@link org.apache.http.HttpStatus#SC_OK}. Also checks the response body
+     * should not be null.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testGetArchiveFile() throws Exception {
+        final Response response = REQUEST.get(FILE_PATH + fileId + "?destinationFormat=plain");
+        final InputStream inputStream = response.body().asInputStream();
+
+        final MessageDigest md = MessageDigest.getInstance("MD5");
+        // Read Content for Length and MD5
+        final CountingInputStream cIn = new CountingInputStream(inputStream);

```

Comment:
Is this used for anything? I see you comparing the md5 but not the length

Reply:
Yes, we are not comparing the file length so CountingInputStream can be removed.  Done.